doc/source/reference/c-api.iterator.rst

```diff
@@ -461,6 +461,33 @@ Construction and Destruction
             Then, call :c:func:`NpyIter_Reset` to allocate and fill the buffers
             with their initial values.
 
+        .. c:var:: NPY_ITER_COPY_IF_OVERLAP
+
+            If a write operand has overlap with a read operand, eliminate all
+            overlap by making temporary copies (with UPDATEIFCOPY for write
+            operands).
+
+            Overlapping means:
+
+            - For a (read, write) pair of operands, there is a memory address
+              that contains data common to both arrays, which can be reached
+              via *different* index/dtype/shape combinations.
+
+            - In particular, unless the arrays have the same shape, dtype,
+              strides, and start address, any shared common data byte accessible
+              by indexing implies overlap.
+
+            Because exact overlap detection has exponential runtime
+            in the number of dimensions, the decision is made based
+            on heuristics, which has false positives (needless copies in unusual
+            cases) but has no false negatives.
+
+            If read/write overlap exists and write operands are modified in the
```

Comment:
I am fine with the write only for now, I think there are some rare cases like `arr -= arr[4]` where it would be a bit better to do the other thing though.